src/Avalonia.Controls/SelectionNode.cs

```diff
@@ -276,12 +283,50 @@ public List<int> SelectedIndices
             }
         }
 
+        public IEnumerable<object> SelectedItems
+        {
+            get => SelectedIndices.Select(x => ItemsSourceView!.GetAt(x));
+        }
+
         public void Dispose()
         {
             ItemsSourceView?.Dispose();
             UnhookCollectionChangedHandler();
         }
 
+        public void BeginOperation()
+        {
+            _changes ??= new SelectionModelChangeSet(this);
+            _changes.BeginOperation();
+
+            for (var i = 0; i < _childrenNodes.Count; ++i)
+            {
+                _childrenNodes[i]?.BeginOperation();
+            }
+        }
+
+        public IEnumerable<SelectionModelChangeSet> EndOperation()
+        {
+            if (_changes != null)
+            {
+                _changes.EndOperation();
+                yield return _changes;
+
+                for (var i = 0; i < _childrenNodes.Count; ++i)
+                {
+                    var child = _childrenNodes[i];
```

Comment:
nit: could use foreach

Reply:
it's iterating the list in reverse order and is removing entries, so, no, it couldn't.

Reply:
Hmm.. how so ? the loop is from 0 to _chlidrennodes.Count...

Reply:
Whoops, got confused by another loop.

Reply:
Yep, you're right. Got too eager with avoiding allocations there I think (enumerating over `IList<>` allocates, but this is a `List<>` so we should be ok).